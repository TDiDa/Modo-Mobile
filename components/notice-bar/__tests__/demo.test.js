import demoTest from '../../../tests/shared/demoTest';

demoTest('notice-bar');
