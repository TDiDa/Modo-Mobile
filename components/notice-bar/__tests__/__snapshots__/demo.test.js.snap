// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ./components/notice-bar/demo/basic.md correctly 1`] = `
<div
  style="padding:10px"
>
  <div
    class="m-notice-bar"
  >
    <div
      class="m-notice-bar-icon"
    >
      <i
        class="m-icon m-icon-notification"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="notification"
          fill="currentColor"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M922 377.7h-57.3v-17.6c0-27-21.3-49-47.4-49H645c-7.2-13.8-18-23.8-32.8-23.8-251.2 0-476.9-217.6-476.9-217.6C128 65.8 120.4 64 113 64c-24.7 0-47.2 19.8-47.2 47.2v801.1c0 27.6 22.6 47.2 47.2 47.2 8 0 16.3-2.1 24-6.7 0 0 208.5-260.4 478.1-260.4 13.8 0 23.7-9.3 30.3-22.2h4.1c1.4 60.3-12.7 124.5-25.8 170.8-7.6 26.9 3 55.2 26.4 70.5 10.6 6.9 22.7 10.4 34.7 10.4 14.4 0 28.7-5 40.5-14.7 69.2-57.5 95.1-123.6 104.7-168.9 6-28.3 6.5-52.2 5.8-68 3.6-4.6 7-9.6 10.3-14.6 18.7-28.7 18.7-39.8 18.7-45.1V588H922c19.9 0 36-16.1 36-36V413.7c0-19.9-16.1-36-36-36zM584.2 621.4c-61.3 3.9-123.6 19.1-185.7 45.4-56.3 23.9-112.5 57-167.1 98.4-37.6 28.5-69.2 57-93.7 81V166.6c26.4 21 61.8 47.1 103.8 73.1 55.2 34.2 110.9 61.6 165.6 81.3 60.2 21.7 119.4 34.3 176.4 37.5v1.7c0 2.3.2 4.5.6 6.6v253.5c.1.4.1.7.1 1.1zm208.5-16.8c-3.1 6.1-9.6 16.1-13.5 21.1-3.2 2-6.1 4.5-8.5 7.4-6.8 8.2-9.7 19-7.8 29.5 0 .2 4.1 25.8-3.3 60.8-8.6 40.8-28.4 77.2-58.9 108.6 14.5-58.5 26.7-133.7 18.1-202.3-2.3-18-17.6-31.5-35.7-31.5h-25.6v-29.9c0-3.3-.5-6.5-1.3-9.5V383.1h136.5v221.5zM886 516h-21.3v-66.3H886V516z"
          />
        </svg>
      </i>
    </div>
    <div
      class="m-notice-bar-content"
    >
      <div
        class="m-notice-bar-marquee-wrap "
        role="marquee"
        style="overflow:hidden"
      >
        <div
          class="m-notice-bar-marquee"
          style="position:relative;right:0;white-space:nowrap;display:inline-block"
        >
          为了确保您的资金安全，请设置支付密码
        </div>
      </div>
    </div>
  </div>
  <br />
  <div
    class="m-notice-bar"
  >
    <div
      class="m-notice-bar-icon"
    >
      <i
        class="m-icon m-icon-notification"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="notification"
          fill="currentColor"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M922 377.7h-57.3v-17.6c0-27-21.3-49-47.4-49H645c-7.2-13.8-18-23.8-32.8-23.8-251.2 0-476.9-217.6-476.9-217.6C128 65.8 120.4 64 113 64c-24.7 0-47.2 19.8-47.2 47.2v801.1c0 27.6 22.6 47.2 47.2 47.2 8 0 16.3-2.1 24-6.7 0 0 208.5-260.4 478.1-260.4 13.8 0 23.7-9.3 30.3-22.2h4.1c1.4 60.3-12.7 124.5-25.8 170.8-7.6 26.9 3 55.2 26.4 70.5 10.6 6.9 22.7 10.4 34.7 10.4 14.4 0 28.7-5 40.5-14.7 69.2-57.5 95.1-123.6 104.7-168.9 6-28.3 6.5-52.2 5.8-68 3.6-4.6 7-9.6 10.3-14.6 18.7-28.7 18.7-39.8 18.7-45.1V588H922c19.9 0 36-16.1 36-36V413.7c0-19.9-16.1-36-36-36zM584.2 621.4c-61.3 3.9-123.6 19.1-185.7 45.4-56.3 23.9-112.5 57-167.1 98.4-37.6 28.5-69.2 57-93.7 81V166.6c26.4 21 61.8 47.1 103.8 73.1 55.2 34.2 110.9 61.6 165.6 81.3 60.2 21.7 119.4 34.3 176.4 37.5v1.7c0 2.3.2 4.5.6 6.6v253.5c.1.4.1.7.1 1.1zm208.5-16.8c-3.1 6.1-9.6 16.1-13.5 21.1-3.2 2-6.1 4.5-8.5 7.4-6.8 8.2-9.7 19-7.8 29.5 0 .2 4.1 25.8-3.3 60.8-8.6 40.8-28.4 77.2-58.9 108.6 14.5-58.5 26.7-133.7 18.1-202.3-2.3-18-17.6-31.5-35.7-31.5h-25.6v-29.9c0-3.3-.5-6.5-1.3-9.5V383.1h136.5v221.5zM886 516h-21.3v-66.3H886V516z"
          />
        </svg>
      </i>
    </div>
    <div
      class="m-notice-bar-content"
    >
      <div
        class="m-notice-bar-marquee-wrap "
        role="marquee"
        style="overflow:hidden"
      >
        <div
          class="m-notice-bar-marquee"
          style="position:relative;right:0;white-space:nowrap;display:inline-block"
        >
          为了确保您的资金安全，请设置支付密码
        </div>
      </div>
    </div>
    <div
      class="m-notice-bar-close-icon"
    >
      <i
        class="m-icon m-icon-close"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="close"
          fill="currentColor"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M567.7 514.9l254.8-254.6c15.4-15.4 15.4-40.3 0-55.6-15.4-15.4-40.3-15.4-55.6 0L512 459.2 257.1 204.6c-15.4-15.4-40.3-15.4-55.6 0-15.4 15.4-15.4 40.3 0 55.6l254.8 254.6-254.8 254.7c-15.4 15.4-15.4 40.3 0 55.6 7.7 7.7 17.8 11.5 27.8 11.5 10.1 0 20.1-3.8 27.8-11.5L512 570.5l254.9 254.7c7.7 7.7 17.7 11.5 27.8 11.5 10.1 0 20.1-3.8 27.8-11.5 15.4-15.4 15.4-40.3 0-55.6L567.7 514.9z"
          />
        </svg>
      </i>
    </div>
  </div>
  <br />
  <div
    class="m-notice-bar"
  >
    <div
      class="m-notice-bar-icon"
    >
      <i
        class="m-icon m-icon-notification"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="notification"
          fill="currentColor"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M922 377.7h-57.3v-17.6c0-27-21.3-49-47.4-49H645c-7.2-13.8-18-23.8-32.8-23.8-251.2 0-476.9-217.6-476.9-217.6C128 65.8 120.4 64 113 64c-24.7 0-47.2 19.8-47.2 47.2v801.1c0 27.6 22.6 47.2 47.2 47.2 8 0 16.3-2.1 24-6.7 0 0 208.5-260.4 478.1-260.4 13.8 0 23.7-9.3 30.3-22.2h4.1c1.4 60.3-12.7 124.5-25.8 170.8-7.6 26.9 3 55.2 26.4 70.5 10.6 6.9 22.7 10.4 34.7 10.4 14.4 0 28.7-5 40.5-14.7 69.2-57.5 95.1-123.6 104.7-168.9 6-28.3 6.5-52.2 5.8-68 3.6-4.6 7-9.6 10.3-14.6 18.7-28.7 18.7-39.8 18.7-45.1V588H922c19.9 0 36-16.1 36-36V413.7c0-19.9-16.1-36-36-36zM584.2 621.4c-61.3 3.9-123.6 19.1-185.7 45.4-56.3 23.9-112.5 57-167.1 98.4-37.6 28.5-69.2 57-93.7 81V166.6c26.4 21 61.8 47.1 103.8 73.1 55.2 34.2 110.9 61.6 165.6 81.3 60.2 21.7 119.4 34.3 176.4 37.5v1.7c0 2.3.2 4.5.6 6.6v253.5c.1.4.1.7.1 1.1zm208.5-16.8c-3.1 6.1-9.6 16.1-13.5 21.1-3.2 2-6.1 4.5-8.5 7.4-6.8 8.2-9.7 19-7.8 29.5 0 .2 4.1 25.8-3.3 60.8-8.6 40.8-28.4 77.2-58.9 108.6 14.5-58.5 26.7-133.7 18.1-202.3-2.3-18-17.6-31.5-35.7-31.5h-25.6v-29.9c0-3.3-.5-6.5-1.3-9.5V383.1h136.5v221.5zM886 516h-21.3v-66.3H886V516z"
          />
        </svg>
      </i>
    </div>
    <div
      class="m-notice-bar-content"
    >
      <div
        class="m-notice-bar-marquee-wrap "
        role="marquee"
        style="overflow:hidden"
      >
        <div
          class="m-notice-bar-marquee"
          style="position:relative;right:0;white-space:nowrap;display:inline-block"
        >
          为了确保您的资金安全，请设置支付密码
        </div>
      </div>
    </div>
    <div
      class="m-notice-bar-close-icon"
    >
      <i
        class="m-icon m-icon-right"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="right"
          fill="currentColor"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M300.6 895.7c9.6 0 19.3-3.2 27.4-9.6l423.8-340.9c10.3-8.3 16.3-20.8 16.3-34.1 0-13.2-6-25.8-16.3-34L329.8 138c-18.8-15.1-46.3-12.1-61.4 6.7-15.1 18.8-12.1 46.3 6.7 61.4l379.6 305.1L273.2 818c-18.8 15.1-21.8 42.6-6.7 61.4 8.7 10.7 21.3 16.3 34.1 16.3z"
          />
        </svg>
      </i>
    </div>
  </div>
</div>
`;

exports[`renders ./components/notice-bar/demo/close-text.md correctly 1`] = `
<div
  style="padding:10px"
>
  <div
    class="m-notice-bar"
  >
    <div
      class="m-notice-bar-icon"
    >
      <i
        class="m-icon m-icon-notification"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="notification"
          fill="currentColor"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M922 377.7h-57.3v-17.6c0-27-21.3-49-47.4-49H645c-7.2-13.8-18-23.8-32.8-23.8-251.2 0-476.9-217.6-476.9-217.6C128 65.8 120.4 64 113 64c-24.7 0-47.2 19.8-47.2 47.2v801.1c0 27.6 22.6 47.2 47.2 47.2 8 0 16.3-2.1 24-6.7 0 0 208.5-260.4 478.1-260.4 13.8 0 23.7-9.3 30.3-22.2h4.1c1.4 60.3-12.7 124.5-25.8 170.8-7.6 26.9 3 55.2 26.4 70.5 10.6 6.9 22.7 10.4 34.7 10.4 14.4 0 28.7-5 40.5-14.7 69.2-57.5 95.1-123.6 104.7-168.9 6-28.3 6.5-52.2 5.8-68 3.6-4.6 7-9.6 10.3-14.6 18.7-28.7 18.7-39.8 18.7-45.1V588H922c19.9 0 36-16.1 36-36V413.7c0-19.9-16.1-36-36-36zM584.2 621.4c-61.3 3.9-123.6 19.1-185.7 45.4-56.3 23.9-112.5 57-167.1 98.4-37.6 28.5-69.2 57-93.7 81V166.6c26.4 21 61.8 47.1 103.8 73.1 55.2 34.2 110.9 61.6 165.6 81.3 60.2 21.7 119.4 34.3 176.4 37.5v1.7c0 2.3.2 4.5.6 6.6v253.5c.1.4.1.7.1 1.1zm208.5-16.8c-3.1 6.1-9.6 16.1-13.5 21.1-3.2 2-6.1 4.5-8.5 7.4-6.8 8.2-9.7 19-7.8 29.5 0 .2 4.1 25.8-3.3 60.8-8.6 40.8-28.4 77.2-58.9 108.6 14.5-58.5 26.7-133.7 18.1-202.3-2.3-18-17.6-31.5-35.7-31.5h-25.6v-29.9c0-3.3-.5-6.5-1.3-9.5V383.1h136.5v221.5zM886 516h-21.3v-66.3H886V516z"
          />
        </svg>
      </i>
    </div>
    <div
      class="m-notice-bar-content"
    >
      <div
        class="m-notice-bar-marquee-wrap "
        role="marquee"
        style="overflow:hidden"
      >
        <div
          class="m-notice-bar-marquee"
          style="position:relative;right:0;white-space:nowrap;display:inline-block"
        >
          Replace the default icon with customized text, long text, long text, long text, long text.
        </div>
      </div>
    </div>
    <div
      class="m-notice-bar-close-icon"
    >
      Close
    </div>
  </div>
</div>
`;

exports[`renders ./components/notice-bar/demo/duration.md correctly 1`] = `
<div
  style="padding:10px"
>
  <div
    class="m-notice-bar"
  >
    <div
      class="m-notice-bar-icon"
    >
      <i
        class="m-icon m-icon-notification"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="notification"
          fill="currentColor"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M922 377.7h-57.3v-17.6c0-27-21.3-49-47.4-49H645c-7.2-13.8-18-23.8-32.8-23.8-251.2 0-476.9-217.6-476.9-217.6C128 65.8 120.4 64 113 64c-24.7 0-47.2 19.8-47.2 47.2v801.1c0 27.6 22.6 47.2 47.2 47.2 8 0 16.3-2.1 24-6.7 0 0 208.5-260.4 478.1-260.4 13.8 0 23.7-9.3 30.3-22.2h4.1c1.4 60.3-12.7 124.5-25.8 170.8-7.6 26.9 3 55.2 26.4 70.5 10.6 6.9 22.7 10.4 34.7 10.4 14.4 0 28.7-5 40.5-14.7 69.2-57.5 95.1-123.6 104.7-168.9 6-28.3 6.5-52.2 5.8-68 3.6-4.6 7-9.6 10.3-14.6 18.7-28.7 18.7-39.8 18.7-45.1V588H922c19.9 0 36-16.1 36-36V413.7c0-19.9-16.1-36-36-36zM584.2 621.4c-61.3 3.9-123.6 19.1-185.7 45.4-56.3 23.9-112.5 57-167.1 98.4-37.6 28.5-69.2 57-93.7 81V166.6c26.4 21 61.8 47.1 103.8 73.1 55.2 34.2 110.9 61.6 165.6 81.3 60.2 21.7 119.4 34.3 176.4 37.5v1.7c0 2.3.2 4.5.6 6.6v253.5c.1.4.1.7.1 1.1zm208.5-16.8c-3.1 6.1-9.6 16.1-13.5 21.1-3.2 2-6.1 4.5-8.5 7.4-6.8 8.2-9.7 19-7.8 29.5 0 .2 4.1 25.8-3.3 60.8-8.6 40.8-28.4 77.2-58.9 108.6 14.5-58.5 26.7-133.7 18.1-202.3-2.3-18-17.6-31.5-35.7-31.5h-25.6v-29.9c0-3.3-.5-6.5-1.3-9.5V383.1h136.5v221.5zM886 516h-21.3v-66.3H886V516z"
          />
        </svg>
      </i>
    </div>
    <div
      class="m-notice-bar-content"
    >
      <div
        class="m-notice-bar-marquee-wrap "
        role="marquee"
        style="overflow:hidden"
      >
        <div
          class="m-notice-bar-marquee"
          style="position:relative;right:0;white-space:nowrap;display:inline-block"
        >
          duration
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ./components/notice-bar/demo/icon.md correctly 1`] = `
<div
  style="padding:10px"
>
  <div
    class="m-notice-bar"
  >
    <div
      class="m-notice-bar-icon"
    >
      <i
        class="m-icon m-icon-question-circle"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="question-circle"
          fill="currentColor"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 0C229.2 0 0 229.2 0 512s229.2 512 512 512 512-229.2 512-512S794.8 0 512 0zm-6.5 812.6c-23.9 0-43.3-19.4-43.3-43.3s19.4-43.3 43.3-43.3 43.3 19.4 43.3 43.3-19.4 43.3-43.3 43.3zM691.3 391c-2.5 46.3-16.7 82.8-87.7 153.8-36 36-58.7 63-61 86.7-1.8 18.6-17.5 32.5-35.8 32.5-1.2 0-2.4-.1-3.6-.2-19.8-1.9-34.2-19.6-32.3-39.4 4.9-49.9 41.5-90.2 81.8-130.5 64.5-64.5 65.6-85 66.7-106.7 1.3-25.3-8-49.8-26.4-69.1-20.9-22-50.5-34.6-81.2-34.6h-.2c-59.1.1-107.3 48.3-107.3 107.5 0 19.9-16.1 36-36 36s-36-16.1-36-36c0-47.9 18.6-92.9 52.4-126.8 33.8-33.9 78.8-52.6 126.7-52.7 50.5-.1 99.3 20.7 133.8 57 32.2 33.7 48.5 77.2 46.1 122.5z"
          />
        </svg>
      </i>
    </div>
    <div
      class="m-notice-bar-content"
    >
      <div
        class="m-notice-bar-marquee-wrap "
        role="marquee"
        style="overflow:hidden"
      >
        <div
          class="m-notice-bar-marquee"
          style="position:relative;right:0;white-space:nowrap;display:inline-block"
        >
          custom icon
        </div>
      </div>
    </div>
  </div>
  <br />
  <div
    class="m-notice-bar"
  >
    <div
      class="m-notice-bar-icon"
    >
      <div
        style="color:#1890ff"
      >
        waring
      </div>
    </div>
    <div
      class="m-notice-bar-content"
    >
      <div
        class="m-notice-bar-marquee-wrap "
        role="marquee"
        style="overflow:hidden"
      >
        <div
          class="m-notice-bar-marquee"
          style="position:relative;right:0;white-space:nowrap;display:inline-block"
        >
          custom text
        </div>
      </div>
    </div>
  </div>
</div>
`;
